### PR TITLE
CLI-80: user-blocks commands with user_id

### PR DESCRIPTION
**Pull request by mhsu-auth0**
### Description

Add commands for managing user-blocks by user_id

Screen Shot 2021-03-26 at 4 51 31 PM

### References

https://auth0team.atlassian.net/browse/CLI-80
https://auth0.com/docs/api/management/v2#!/User_Blocks/delete_user_blocks

https://github.com/go-auth0/auth0/blob/master/management/user.go#L423-L449

### CODE REVIEW

File: internal/cli/user_blocks.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+var (
+	userID = Argument{
+		Name: "user_id",
+		Help: "user_id of the user.",
+	}
+)
+
+func userBlocksCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "user-blocks",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:21]: Please put it under `users`

File: internal/cli/user_blocks.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"fmt"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+	"github.com/spf13/cobra"
+	"gopkg.in/auth0.v5/management"
+)
+
+var (
+	userID = Argument{
+		Name: "user_id",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:14]:
```suggestion
		Name: "User ID",
```

File: internal/cli/user_blocks.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		Help: "user_id of the user.",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:15]:
```suggestion
		Help: "Id of the user.",
```

File: internal/cli/user_blocks.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+	cmd.SetUsageTemplate(resourceUsageTemplate())
+	cmd.AddCommand(listUserBlocksByUserIdCmd(cli))
+	cmd.AddCommand(deleteUserBlocksByUserIdCmd(cli))
+	return cmd
+}
+
+func listUserBlocksByUserIdCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		user_id string
```
Comment (Widcket) [inline internal/cli/user_blocks.go:33]: Please do not use snake_case

File: internal/cli/user_blocks.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	}
+
+	cmd := &cobra.Command{
+		Use:   "listByUserId",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:37]: Please use `list`. Entire command should be `auth0 users blocks list`.

Comment (mhsu-auth0) [reply internal/cli/user_blocks.go:37]: we can list by `user_id` or list by `email / phonenumber / user_identifier`. It's a different API call, what would a good way to name the handle the two different input types?

File: internal/cli/user_blocks.go (hunk 1 above)
Comment (Widcket) [reply internal/cli/user_blocks.go:21]: As a subcommand: `auth0 users blocks`

File: internal/cli/user_blocks.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+		Args:  cobra.MaximumNArgs(1),
+		Short: "List user-blocks by user_id",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:39]: Please do not include the API2 entities as-is, instead describe what the command does in the context of a single user.

File: internal/cli/user_blocks.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+		Long: `List user-blocks by user_id:
```
Comment (Widcket) [inline internal/cli/user_blocks.go:40]: Same as above.

File: internal/cli/user_blocks.go [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func deleteUserBlocksByUserIdCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		user_id string
```
Comment (Widcket) [inline internal/cli/user_blocks.go:76]: Please do not use snake_case.

File: internal/cli/user_blocks.go [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+	}
+
+	cmd := &cobra.Command{
+		Use:   "deleteByUserId",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:80]: Please use`delete`. Entire command should be `auth0 users blocks delete`.

File: internal/cli/user_blocks.go [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Delete all user-blocks by user_id",
```
Comment (Widcket) [inline internal/cli/user_blocks.go:82]: Please do not include the API2 entities as-is, instead describe what the command does in the context of a single user.

File: internal/cli/user_blocks.go [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+		Long: `Delete all user-blocks by user_id:
```
Comment (Widcket) [inline internal/cli/user_blocks.go:83]: Sma as above.

File: internal/display/user_blocks.go [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+func (v *userBlockView) KeyValues() [][]string {
+	return [][]string{
+		[]string{"Identifier", v.Identifier},
+		[]string{"IP", v.IP},
+	}
+}
+
+func (r *Renderer) UserBlocksList(userBlocks []*management.UserBlock) {
+	resource := "userBlocks"
```
Comment (Widcket) [inline internal/display/user_blocks.go:28]:
```suggestion
	resource := "user blocks"
```

File: internal/display/user_blocks.go [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+	r.Heading(resource)
+
+	if len(userBlocks) == 0 {
+		r.EmptyState(resource)
+		r.Infof("No blocks for user.")
```
Comment (Widcket) [inline internal/display/user_blocks.go:34]: `r.EmptyState(resource)` and `r.Infof("No blocks for user.")` do the same thing, please remove the last one.

**Review by Widcket — CHANGES_REQUESTED** [review]
Please refactor to fit in with the rest of the CLI. Please use the `apps`, `apis`, and `rules` commands as a reference.

File: internal/cli/user_blocks.go (hunk 5 above)
Comment (Widcket) [reply internal/cli/user_blocks.go:37]: The command should accept any of these as a single argument, and then disambiguate. It's the easiest for the user, that remains unaware (as they should) of those implementation details.

Comment (Widcket) [reply internal/cli/user_blocks.go:37]: e.g. `auth0 apis show` accepts an API id or the audience.

Comment (Widcket) [reply internal/cli/user_blocks.go:37]: For the end user, there's no difference.

File: internal/cli/user_blocks.go (hunk 10 above)
Comment (Widcket) [reply internal/cli/user_blocks.go:82]: Assume the user knows nothing about API2.

File: internal/cli/user_blocks.go (hunk 6 above)
Comment (Widcket) [reply internal/cli/user_blocks.go:39]: Assume the user knows nothing about API2.

**Comment by mhsu-auth0** [comment]
moving to branch on main repo

**Pull request closed without merging**